google/gcloud/config.go

```diff
@@ -1,55 +1,36 @@
 package gcloud
 
+import (
+	"github.com/pkg/errors"
+	"github.com/samber/lo"
+)
+
 type Config struct {
-	ConfigDir    string        `json:"configDir" yaml:"configDir"`
-	Environments []Environment `json:"environments" yaml:"environments"`
+	ConfigPath string             `json:"configPath" yaml:"configPath"`
+	Accounts   map[string]Account `json:"accounts" yaml:"accounts"`
+	Clusters   map[string]Cluster `json:"clusters" yaml:"clusters"`
 }
 
-func (c Config) FindCluster(envName, clusterName string) (Cluster, bool) {
-	for _, env := range c.Environments {
-		if env.Name != envName {
-			continue
-		}
-
-		for _, cluster := range env.Clusters {
-			if cluster.Name == clusterName {
-				return cluster, true
-			}
-		}
+func (c Config) Cluster(name string) (Cluster, error) {
+	if value, ok := c.Clusters[name]; !ok {
+		return Cluster{}, errors.Errorf("given cluster not found: %s", name)
+	} else {
+		return value, nil
 	}
-	return Cluster{}, false
-}
-
-type Environment struct {
-	Name     string    `json:"name" yaml:"name"`
-	Clusters []Cluster `json:"clusters" yaml:"clusters"`
 }
 
 func (c Config) ClusterNames() []string {
-	var ret []string
-	for _, account := range c.Environments {
-		for _, cluster := range account.Clusters {
-			ret = append(ret, cluster.Name)
-		}
-	}
-	return ret
+	return lo.Keys(c.Clusters)
 }
 
-func (c Config) ClusterNamesForEnv(envName string) []string {
-	for _, env := range c.Environments {
-		if env.Name == envName {
-			names := make([]string, len(env.Clusters))
-			for idx, cluster := range env.Clusters {
-				names[idx] = cluster.Name
-			}
-			return names
-		}
+func (c Config) Account(name string) (Account, error) {
+	if value, ok := c.Accounts[name]; !ok {
```

Comment:
else statement is not required 
https://go.dev/doc/effective_go#if
it's recommended to be omitted